conda/gateways/connection/session.py

```diff
@@ -70,21 +132,30 @@ def __new__(mcs, name, bases, dct):
         dct["_thread_local"] = local()
         return super().__new__(mcs, name, bases, dct)
 
-    def __call__(cls):
+    def __call__(cls, **kwargs):
+        storage_key = get_session_storage_key(kwargs.get("auth"))
+
         try:
-            return cls._thread_local.session
+            return cls._thread_local.sessions[storage_key]
         except AttributeError:
-            session = cls._thread_local.session = super().__call__()
-            return session
+            session = super().__call__(**kwargs)
+            cls._thread_local.sessions = {storage_key: session}
+        except KeyError:
+            session = cls._thread_local.sessions[storage_key] = super().__call__(
+                **kwargs
+            )
+
+        return session
 
 
 class CondaSession(Session, metaclass=CondaSessionType):
-    def __init__(self):
+    def __init__(self, auth: AuthBase | None = None):
```

Comment:
Since we add a check in `get_session_storage_key()` to see if `auth` is a tuple, should we update the type hinting here to include that possibility?
```suggestion
    def __init__(self, auth: AuthBase | tuple[str, str] | None = None):
```